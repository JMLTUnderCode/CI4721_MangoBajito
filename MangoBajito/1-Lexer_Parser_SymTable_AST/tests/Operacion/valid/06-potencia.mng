# Test 06: Operacion de potencia.
se_prende(){
	culito n0: mango = -2;
	n0 = n0 ** 3;
	n0 = n0 ** 2 ** 2;
	n0 = (3 ** 3) ** (2 ** 2);

	culito n1: manguita = -2.9;
	n1 = n1 ** 2.2;
	n1 = n1 ** 2.7 ** 2.12;
	n1 = (3.2 ** 1.23) ** (2.3 ** 2.22);

	culito n2: manguangua = 1.2e+2;
	n2 = n2 ** 0.022e+2;
	n2 = n2 ** 0.021e+2 ** 0.031e+2;
	n2 = (0.022e+2 ** 0.025e+2) ** (0.0201e+2 ** 0.02e+2);
};