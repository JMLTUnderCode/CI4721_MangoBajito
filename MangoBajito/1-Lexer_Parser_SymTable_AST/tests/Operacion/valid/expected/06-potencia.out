| Acom√≥date que te lo estan analizando d:-)
--> Print table? (s/n): [1;36m[5m
               =======================================================
                             Symbol Table Representation              
               =======================================================
[0m
[1;35m[5m    Clave: 1: n2[0m
    |---> S√≠mbolo: n2, Categor√≠a: (1)VARIABLE, Scope: 2, Type: manguangua, Value: 2874.45

[1;35m[5m    Clave: 2: n1[0m
    |---> S√≠mbolo: n1, Categor√≠a: (1)VARIABLE, Scope: 2, Type: manguita, Value: 8868.61

[1;35m[5m    Clave: 3: n0[0m
    |---> S√≠mbolo: n0, Categor√≠a: (1)VARIABLE, Scope: 2, Type: mango, Value: 531441

[1;35m[5m    Clave: 4: mango[0m
    |---> S√≠mbolo: mango, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 5: manguita[0m
    |---> S√≠mbolo: manguita, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 6: manguangua[0m
    |---> S√≠mbolo: manguangua, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 7: negro[0m
    |---> S√≠mbolo: negro, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 8: tas_claro[0m
    |---> S√≠mbolo: tas_claro, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 9: hablame[0m
    |---> S√≠mbolo: hablame, Categor√≠a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 10: un_co√±o[0m
    |---> S√≠mbolo: un_co√±o, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 11: array$[0m
    |---> S√≠mbolo: array$, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 12: funcion$[0m
    |---> S√≠mbolo: funcion$, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 13: error$[0m
    |---> S√≠mbolo: error$, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 14: higuerote[0m
    |---> S√≠mbolo: higuerote, Categor√≠a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 15: rescata[0m
    |---> S√≠mbolo: rescata, Categor√≠a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 16: se_prende[0m
    |---> S√≠mbolo: se_prende, Categor√≠a: (2)FUNCTION, Scope: 0

--> Print AST? (s/n): [1;36m[5m
               =======================================================
                                   AST Representation                 
               =======================================================
[0m
Programa
|--> Main
     |--> Instrucci√≥n | Kind: ;
          |--> Instrucci√≥n | Kind: ;
          |    |--> Instrucci√≥n | Kind: ;
          |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |    |    |    |    |--> Instrucci√≥n | Kind: ;
          |    |    |    |    |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |    |    |    |    |    |    |--> n0 | Category: Declaraci√≥n | Type: mango | Kind: VARIABLE
          |    |    |    |    |    |    |    |    |    |    |    |--> Literal | Category: Num√©rico | Type: mango | Value: -2
          |    |    |    |    |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |    |    |    |    |         |--> n0 | Category: Identificador | Type: mango
          |    |    |    |    |    |    |    |    |    |         |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: -8.000000
          |    |    |    |    |    |    |    |    |    |              |--> n0 | Category: Identificador | Type: mango | Value: -2
          |    |    |    |    |    |    |    |    |    |              |--> Literal | Category: Num√©rico | Type: mango | Value: 3
          |    |    |    |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |    |    |    |         |--> n0 | Category: Identificador | Type: mango
          |    |    |    |    |    |    |    |    |         |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: 4096.000000
          |    |    |    |    |    |    |    |    |              |--> n0 | Category: Identificador | Type: mango | Value: -8
          |    |    |    |    |    |    |    |    |              |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: 4.000000
          |    |    |    |    |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: mango | Value: 2
          |    |    |    |    |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: mango | Value: 2
          |    |    |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |    |    |         |--> n0 | Category: Identificador | Type: mango
          |    |    |    |    |    |    |    |         |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: 531441.000000
          |    |    |    |    |    |    |    |              |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: 27.000000
          |    |    |    |    |    |    |    |              |    |--> Literal | Category: Num√©rico | Type: mango | Value: 3
          |    |    |    |    |    |    |    |              |    |--> Literal | Category: Num√©rico | Type: mango | Value: 3
          |    |    |    |    |    |    |    |              |--> ** | Category: Operaci√≥n | Type: mango | Kind: Num√©rica | Value: 4.000000
          |    |    |    |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: mango | Value: 2
          |    |    |    |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: mango | Value: 2
          |    |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |    |         |--> n1 | Category: Declaraci√≥n | Type: manguita | Kind: VARIABLE
          |    |    |    |    |    |    |         |--> Literal | Category: Num√©rico | Type: manguita | Value: -2.900000
          |    |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |    |         |--> n1 | Category: Identificador | Type: manguita
          |    |    |    |    |    |         |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: -10.405805
          |    |    |    |    |    |              |--> n1 | Category: Identificador | Type: manguita | Value: -2.900000
          |    |    |    |    |    |              |--> Literal | Category: Num√©rico | Type: manguita | Value: 2.200000
          |    |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |    |         |--> n1 | Category: Identificador | Type: manguita
          |    |    |    |    |         |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: -226297472.000000
          |    |    |    |    |              |--> n1 | Category: Identificador | Type: manguita | Value: -10.405805
          |    |    |    |    |              |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: 8.212798
          |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: manguita | Value: 2.700000
          |    |    |    |    |                   |--> Literal | Category: Num√©rico | Type: manguita | Value: 2.120000
          |    |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |    |         |--> n1 | Category: Identificador | Type: manguita
          |    |    |    |         |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: 8868.611328
          |    |    |    |              |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: 4.181523
          |    |    |    |              |    |--> Literal | Category: Num√©rico | Type: manguita | Value: 3.200000
          |    |    |    |              |    |--> Literal | Category: Num√©rico | Type: manguita | Value: 1.230000
          |    |    |    |              |--> ** | Category: Operaci√≥n | Type: manguita | Kind: Num√©rica | Value: 6.353833
          |    |    |    |                   |--> Literal | Category: Num√©rico | Type: manguita | Value: 2.300000
          |    |    |    |                   |--> Literal | Category: Num√©rico | Type: manguita | Value: 2.220000
          |    |    |    |--> Asignaci√≥n | Kind: =
          |    |    |         |--> n2 | Category: Declaraci√≥n | Type: manguangua | Kind: VARIABLE
          |    |    |         |--> Literal | Category: Num√©rico | Type: manguangua | Value: 1.2000000000e+02
          |    |    |--> Asignaci√≥n | Kind: =
          |    |         |--> n2 | Category: Identificador | Type: manguangua
          |    |         |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 3.7514463620e+04
          |    |              |--> n2 | Category: Identificador | Type: manguangua | Value: 1.2000000000e+02
          |    |              |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.2000000000e+00
          |    |--> Asignaci√≥n | Kind: =
          |         |--> n2 | Category: Identificador | Type: manguangua
          |         |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 4.2087780119e+45
          |              |--> n2 | Category: Identificador | Type: manguangua | Value: 3.7514463620e+04
          |              |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 9.9742399927e+00
          |                   |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.1000000000e+00
          |                   |--> Literal | Category: Num√©rico | Type: manguangua | Value: 3.1000000000e+00
          |--> Asignaci√≥n | Kind: =
               |--> n2 | Category: Identificador | Type: manguangua
               |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 2.8744494925e+03
                    |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 7.1788801355e+00
                    |    |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.2000000000e+00
                    |    |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.5000000000e+00
                    |--> ** | Category: Operaci√≥n | Type: manguangua | Kind: Num√©rica | Value: 4.0401000000e+00
                         |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.0100000000e+00
                         |--> Literal | Category: Num√©rico | Type: manguangua | Value: 2.0000000000e+00
[1;32m[5m
               =======================================================
                       ---->       Correct Program       <----        
               =======================================================
[0m
| Hay un poquito de c√≥digo en tu marihuana d:-]
